File: javascript01/step11/test02_1.js
/* 데이터 조회 명령: select ~ from ~
select 컬럼명, 컬럼명, 컬럼명....
from 테이블명, 테이블명, 테이블명....
Where
*/



const mysql  = require('mysql');
const connection = mysql.createConnection({

  database: 'webappdb',
  user: 'webapp',
  password: '1111'
})


connection.connect()


connection.query('select mno, name, tel  from memb', function(error, results) {
  console.log('결과를 가져왔음!')
  if (error) throw error;

  for (var row of results) {
    console.log(row.mno, row.name, row.tel)
  }
})

connection.end()
console.log('MySQL 서버와의 연결 끊기가 예약되었음!')

File: java01/src/step20/Test02_1.java
/* JDBC 프로그래밍: Statement 객체
 * => SQL 문을 DBMS에 전달하고 ,그 결과를 다루는 객체를 리턴한다.
 *  executeQuery() : SELECT문 실행
 *  executeUpdate(): INSERT, UPDATE, DELETE
 *  executeBatch() : SELECT, INSERT, UPDATE, DELETE
 */
package step20;

import java.sql.DriverManager;

public class Test02_1 {

	public static void main(String[] args) throws Exception {
		
		com.mysql.jdbc.Driver mysqldriver = new com.mysql.jdbc.Driver();
		
		DriverManager.registerDriver(mysqldriver); // close() 없음
		
		java.sql.Connection con = DriverManager.getConnection(
				"jdbc:mysql://localhost:3306/webappdb", 
				"webapp",
				"1111");
		
		java.sql.Statement stmt = con.createStatement();
		
		java.sql.ResultSet rs = stmt.executeQuery("select mno, name, tel, email from memb");
		
		while (rs.next()) {
			System.out.printf("%d, %s, %s, %s\n",
					rs.getInt("mno"),
					rs.getString("name"),
					rs.getString("tel"),
					rs.getString("email"));
		}
		rs.close();
		
		stmt.close();
		
		con.close();
		
		
	}
	
}
